src/main/java/seedu/multitasky/logic/parser/AddCommandParser.java

```diff
@@ -22,32 +29,108 @@
     /**
      * Parses the given {@code String} of arguments in the context of the
      * AddCommand and returns an AddCommand object for execution.
-     *
      * throws ParseException if the user input does not conform the expected format.
      */
     public AddCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TAG);
-        String trimmedArgs = argMultimap.getPreamble().get();
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_BY, PREFIX_AT, PREFIX_FROM,
+                                                 PREFIX_TO, PREFIX_TAG);
+        Calendar startDate = Calendar.getInstance();
+        Calendar endDate = Calendar.getInstance();
 
-        if (trimmedArgs.isEmpty()) {
+        // check for no args input
+        if (args.trim().isEmpty()) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+                                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
```

Comment:
is this from eclipse? doesn't really make sense compared to just having it on the previous line